### PR TITLE
Refactor: use kv internally for docstate

### PR DESCRIPTION
**Pull request by zxch3n**

### CODE REVIEW

File: crates/loro-internal/src/state/container_store/container_wrapper.rs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+    pub fn new_from_bytes(b: Bytes) -> Self {
+        let src: &[u8] = &b;
+        let bytes: &[u8] = &b;
+        let kind = ContainerType::try_from_u8(bytes[0]).unwrap();
+        let mut bytes = &bytes[1..];
+        let depth = leb128::read::unsigned(&mut bytes).unwrap();
+        let (parent, bytes) = postcard::take_from_bytes(bytes).unwrap();
+        // SAFETY: bytes is a slice of b
+        let size = unsafe { bytes.as_ptr().offset_from(src.as_ptr()) };
```
Comment (Leeeon233) [inline crates/loro-internal/src/state/container_store/container_wrapper.rs:133]: Is it not equal `src.len() - bytes.len()`?

Comment (zxch3n) [reply crates/loro-internal/src/state/container_store/container_wrapper.rs:133]: You're right!

Comment (zxch3n) [reply crates/loro-internal/src/state/container_store/container_wrapper.rs]: Fixed

File: crates/loro-internal/src/state/container_store/container_wrapper.rs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                self.value = Some(v);
+                self.bytes_offset_for_state = Some(0);
+                return Ok(());
+            }
+            ContainerType::Unknown(_) => UnknownState::decode_value(b)?,
+        };
+
+        self.value = Some(v);
+        // SAFETY: rest is a slice of b
+        let offset = unsafe { rest.as_ptr().offset_from(b.as_ptr()) };
```
Comment (Leeeon233) [inline crates/loro-internal/src/state/container_store/container_wrapper.rs:204]: here

**Review by Leeeon233 — APPROVED** [review]

**Pull request merged**